### PR TITLE
fix(#8541): fixes display of Enketo select option

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

- Fixes display for:
  - Long and shot words
  - Long and short sentences 

medic/cht-core#8541

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8541_long_names_enketo/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8541_long_names_enketo/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8541_long_names_enketo/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@garethbowen can you please review this quick fix? Thanks

**Comment by garethbowen** [comment]
It looks like the width has broken radio buttons:

image

Have you got a form with a select that you can share so I can test this out?

**Comment by latin-panda** [comment]
@garethbowen I used the [enketo widget form](https://github.com/medic/support-scripts/tree/master/forms/enketo_widgets/forms/app)

**Comment by latin-panda** [comment]
@garethbowen please hold, I'm checking some input display

**Comment by garethbowen** [comment]
@latin-panda How are you getting on with this one?

**Comment by latin-panda** [comment]
It turned out to be more challenging to fix than I expected, but I think it's okay now.

Testing using [enketo widget form](https://github.com/medic/support-scripts/tree/master/forms/enketo_widgets/forms/app)

<details> 
<summary>Short text</summary>





</details>

<details> 
<summary>Long text</summary>







</details>

<details> 
<summary>Just normal text</summary>




</details>

@garethbowen can you please review?

**Comment by tatilepizs** [comment]
Hi @latin-panda,

I was testing your branch and as you said, everything is looking fine with the `enketo_widgets` form, but the other forms that contain translations display all the available labels. Could you please check what might be wrong?

<details>
<summary>8541_long_names_enketo</summary>

image
image

</details>

<details>
<summary>master</summary>

image
image



</details>

File: webapp/src/css/enketo/medic.less [hunk 1]
```diff
@@ -18,10 +18,21 @@
...
+    }
+
+    .question:not(.or-appearance-likert) .option-wrapper {
+      padding: 5px;
       .option-label {
         white-space: normal;
-        word-break: normal;
+        word-break: break-all;
+        margin: 0;
+        display: contents;
```
Comment (garethbowen) [inline webapp/src/css/enketo/medic.less:35]: I suspect this is causing the bug that @tatilepizs found - translations probably have `display: none` and this is overwriting them.

**Review by garethbowen — CHANGES_REQUESTED** [review]
Overall looks good, but rejected based on the bug.

File: webapp/src/css/enketo/medic.less (hunk 1 above)
Comment (tatilepizs) [reply webapp/src/css/enketo/medic.less:35]: I commented on that line and all the translation go away, but this is how it looks:

image

Comment (latin-panda) [reply webapp/src/css/enketo/medic.less:35]: Fixed! Thank you!

**Comment by latin-panda** [comment]
@tatilepizs I've fixed it.
Inputs with dropdowns and long text behave like in `master` branch (no wrapping text), but the original issue is solved. The reason is that I don't have enough uniqueness (classes, ids, etc) in the Enketo's HTML to isolate the change without affecting other fields. 

<details> 
<summary>Screenshots</summary>

Screenshot 2023-10-24 at 5 02 41 pm
Screenshot 2023-10-24 at 5 02 15 pm
Screenshot 2023-10-24 at 5 02 56 pm
Screenshot 2023-10-24 at 8 43 11 pm
Screenshot 2023-10-24 at 8 42 53 pm
Screenshot 2023-10-24 at 8 42 36 pm
</details>

**Review by tatilepizs — APPROVED** [review]
Thank you @latin-panda for fixing it so quickly ✨ 

I just have a question that is definitely not a blocker. 
Why for some elements it is going to the next line in the correct place (blank space) and in other elements is cutting the word? 
Is this something related to the code or is more related to the form? Do you know?

For example:

<details>
<summary>Places when the sentence/word is cut in the correct way:</summary>

image
image
image

</details>

<details>
<summary>Places when the sentence/word is cut in the wrong way:</summary>

image
image
image

</details>

File: webapp/src/css/enketo/medic.less [hunk 2]
```diff
@@ -18,10 +18,21 @@
...
+    }
+
+    .question:not(.or-appearance-likert):not(.or-appearance-columns) .option-wrapper {
+      padding: 5px;
+      .option-label.active {
         white-space: normal;
-        word-break: normal;
+        word-break: break-all;
+        margin: 0;
+        display: contents;
```
Comment (garethbowen) [inline webapp/src/css/enketo/medic.less:35]: I don't think `display: contents` is valid in this context. I get quite a nice result if I remove these two lines altogether...

```suggestion
```

Your PR:

Screenshot from 2023-10-25 12-19-12

My suggestion:

Screenshot from 2023-10-25 12-19-02

File: webapp/src/css/enketo/medic.less [hunk 3]
```diff
@@ -18,10 +18,21 @@
...
+
+    .dropdown-menu .option-wrapper {
+      padding: 3px;
       .option-label {
+        width: calc(100% - 34px);
+      }
+    }
+
+    .question:not(.or-appearance-likert):not(.or-appearance-columns) .option-wrapper {
+      padding: 5px;
```
Comment (garethbowen) [inline webapp/src/css/enketo/medic.less:30]: I think I prefer it without this padding as it means we can use 10px more width which is useful on mobile. Why is this needed?

File: webapp/src/css/enketo/medic.less [hunk 4]
```diff
@@ -434,8 +445,7 @@
     .option-wrapper > label:not(.filler) {
       padding-top: 9px;
       padding-bottom: 9px;
-      margin-top: 1px;
-      margin-bottom: 1px;
+      margin: 0;
```
Comment (garethbowen) [inline webapp/src/css/enketo/medic.less:448]: You've set `margin: 0` on the desktop version so we don't need to overwrite it in the mobile version. I think you can just drop this line altogether.

```suggestion
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Starting to look really good, thanks!

I've left a few suggestions inline.

Also in your screenshot here the dropdown option label isn't wrapping at all. I didn't manage to figure out what was going on but maybe you can see something?

https://user-images.githubusercontent.com/66472237/277677449-dad4ca04-b6e0-47dc-96ed-3848e0fbd53d.png

**Comment by latin-panda** [comment]
> Also in your screenshot here the dropdown option label isn't wrapping at all. I didn't manage to figure out what was going on but maybe you can see something?

@garethbowen That's what I tried to explain [here](https://github.com/medic/cht-core/pull/8565#issuecomment-1777244358), I didn't find a way to wrap it, and that doesn't affect other fields display

File: webapp/src/css/enketo/medic.less (hunk 2 above)
Comment (latin-panda) [reply webapp/src/css/enketo/medic.less:35]: We need the ` display: contents;` so radios and checkboxes utilize the space efficiently and display nicely when in columns. 
This is _without_ ` display: contents;`
Screenshot 2023-10-25 at 11 50 24 am
This is _with_ ` display: contents;`
Screenshot 2023-10-25 at 11 52 40 am

File: webapp/src/css/enketo/medic.less (hunk 3 above)
Comment (latin-panda) [reply webapp/src/css/enketo/medic.less:30]: I tried to fix something with that, but I can't remember anymore nor reproduce anything weird by not having it. 
I'm just going to remove it.

File: webapp/src/css/enketo/medic.less (hunk 2 above)
Comment (latin-panda) [reply webapp/src/css/enketo/medic.less:35]: I've removed the margin

**Comment by latin-panda** [comment]
> Why for some elements it is going to the next line in the correct place (blank space) and in other elements is it cutting the word? Is this something related to the code or is more related to the form? Do you know?

@tatilepizs somewhere else in the code, there's an additional rule to break by word instead of by letter when the field is in a column. I initially made it by letter in case of weird big words, which may be uncommon. It looks like the browser deals with long words just well, so I changed it to break by word. Thanks for the observation!

File: webapp/src/css/enketo/medic.less (hunk 2 above)
Comment (garethbowen) [reply webapp/src/css/enketo/medic.less]: Ahh ok, thanks.

**Review by garethbowen — APPROVED** [review]
Thanks for going back and forwards on this!

**Comment by garethbowen** [comment]
This is so complicated and requires so much testing I wonder if we need a test suite which compares screenshots. Then we can create a form with all of these edge cases and automatically check that we haven't regressed somewhere else...

**Comment by dianabarsan** [comment]
This change ended up making all label elements not interact able in new wdio, and also in chrome dev tools - if I open dev tools and click on an element, it won't get focused :(

**Comment by dianabarsan** [comment]
Related chrome issue: https://bugs.chromium.org/p/chromium/issues/detail?id=1366037
I got this link from this page: https://caniuse.com/css-display-contents#:~:text=display%3A%20contents%20causes%20an%20element's,grid%20or%20similar%20layout%20techniques.
